github-copilot-case-study.md

```diff
@@ -0,0 +1,113 @@
+
+## Brief History
+
+Github Copilot is a collaboration between OpenAI and Github, first launched as a technical preview for visual studio code in june 2021. Support for JetBrains and Neovim came in october same year. Support for visual studio came in march of 2022. In june of 2022 Github announced that Copilot was out of beta and available for individual developers as a subscription service at 10$/month. In the middle of february 2023 Github launched Copilot for business, adding some additional features and priced at 19$/month per user.
+
+## Features
+
+Github Copilot offers a wide range of features to help developers write code faster and easier. The main features are:
+
+- Code completion
+- Code suggestions
+- Code generation
+
+In addition to that Github Copilot can also fill in boilerplate code like building a simple html page using just a few prompts to describe the page, It can fill in placeholder copy for example if you find lorem ipsum to be a bit to repetitive. One example of this is this document you are reading right now, as I am writing this Copilot is giving me real time suggestions and completions for the text I am writing and doing a surprisingly good job at it.
+
+## Strengths
+
+I think one of Copilot's biggest strengths is that it is really fast, the suggestions come almost instantly as I am writing and from my testing so far it doesn't seem to take a big toll on system performance. Other strengths include:
+
+- Easy IDE integration
+- Ok [pricing](https://copilot.github.com/pricing)
+- Easy to use and very intuitive
+- Good language support for most of the most common languages
+- Good code suggestions and completions
+- Possibility to use either to complete code or to generate new code based on comments
+- Easy to install and setup
+- 60 Days free trial
+
+## Weaknesses
+
+I would say one of it's biggest weaknesses is that it's not perfect. While the code suggestions and completions are really good they're not perfect and you will still have to read trough every line of code to make sure it's all correct and does what you expect. Another weakness is that it's not free. In my opinion the pricing is ok and seems worth it so far for me at least but I can see how some people might not be willing to pay for it. It also shares a weakness i've heard a lot of discussion about with chatGPT and other natural language models, that it can be wrong very convincingly. This is especially true when writing, like when writing this document. I've seen several suggestions that are completely wrong, like making suggestions to write that Copilot is free, that it support languages or ide's that it doesn't support and it keeps suggesting that it's still in beta. For instance, Copilot suggested this sentence to follow the last:
+
+"I've also seen it suggest that it's not free and that it's not in beta. So it's not perfect and it can be wrong but it's still really good and I think it's worth the price."
```

Comment:
Is this a quote?

```suggestion
> "I've also seen it suggest that it's not free and that it's not in beta. So it's not perfect and it can be wrong but it's still really good and I think it's worth the price."
```

Reply:
Yes that is a quote  of the sentence copilot suggested to come after this: "I've seen several suggestions that are completely wrong, like making suggestions to write that Copilot is free, that it support languages or ide's that it doesn't support and it keeps suggesting that it's still in beta."

Reply:
Gotcha, quotes are marked with `>`

Reply:
Oh ok, I'll merge the change then